tests/backup/backup_resiliency_test.go

```diff
@@ -758,3 +761,253 @@ var _ = Describe("{CancelAllRunningBackupJobs}", func() {
+			log.Infof("Verify that at least one mongodb pod is in Ready state")
+			mongoDBPodStatus := func() (interface{}, bool, error) {
+				statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, backupNS)
+				if err != nil {
+					return "", true, fmt.Errorf("unable to get the mongodb statefulset")
+				}
+				if statefulSet.Status.ReadyReplicas < 1 {
+					return "", true, fmt.Errorf("no mongodb pods are ready yet")
+				}
+				return "", false, nil
+			}
+			_, err = task.DoRetryWithTimeout(mongoDBPodStatus, 5*time.Minute, 30*time.Second)
+			log.Infof("Number of mongodb pods in Ready state are %v", statefulSet.Status.ReadyReplicas)
+			dash.VerifyFatal(statefulSet.Status.ReadyReplicas > 0, true, "Verifying that at least one mongodb pod is in Ready state")
+		})
+		Step("Check if backup is successful after MongoDB statefulset is scaled back to original replica", func() {
+			log.InfoD("Check if backup is successful after MongoDB statefulset is scaled back to original replica")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				backupStatus, err := backupSuccessCheck(backupName, orgID, 0, 0, ctx)
+				dash.VerifyFatal(err, nil, "Getting the status for backup- "+backupName)
+				dash.VerifyFatal(backupStatus, true, "Verifying the backup status for backup - "+backupName)
+			}
+		})
+		Step("Restoring the backups taken", func() {
+			log.InfoD("Restoring the backups taken")
+			ctx, err = backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting admin context")
+			for _, backupName := range backupNames {
+				sem <- struct{}{}
+				restoreName := fmt.Sprintf("%s-restore", backupName)
+				restoreNames = append(restoreNames, restoreName)
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					defer func() { <-sem }()
+					_, err = CreateRestoreWithoutCheck(restoreName, backupName, nil, destinationClusterName, orgID, ctx)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
+				}(backupName)
+			}
+			wg.Wait()
+			log.Infof("The list of restores are: %v", restoreNames)
+		})
+		Step("Scaling MongoDB statefulset replica to 0 and back to original replica while restore is in progress", func() {
+			log.InfoD("Scaling MongoDB statefulset replica to 0 while restore is in progress")
+			*statefulSet.Spec.Replicas = scaledDownReplica
+			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.Infof("mongodb replica after scaling to 0 is %v", *statefulSet.Spec.Replicas)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
+			dash.VerifyFatal(*statefulSet.Spec.Replicas == scaledDownReplica, true, "Getting mongodb statefulset replica after scaling down")
```

Comment:
Wouldn't it be better if we retrieve the spec replicas using GetStatefulSet and compare it instead?

Reply:
I apologize for my oversight. I did not notice that UpdateStatefulSet is also returning a statefulset.